File: GXpert/GXpert.Web/Modules/ServerTypes/LiveSessions/LiveSessionLogRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib";

export interface LiveSessionLogRow {
    Id?: number;
    Date?: string;
    StartTime?: string;
    EndTime?: string;
    TeacherId?: number;
    ClassId?: number;
    SubjectId?: number;
    InsertDate?: string;
    InsertUserId?: number;
    UpdateDate?: string;
    UpdateUserId?: number;
    IsActive?: boolean;
    TeacherPrn?: string;
    ClassTitle?: string;
    SubjectTitle?: string;
}

export abstract class LiveSessionLogRow {
    static readonly idProperty = 'Id';
    static readonly localTextPrefix = 'LiveSessions.LiveSessionLog';
    static readonly lookupKey = 'LiveSessions.LiveSessionLog';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<LiveSessionLogRow>('LiveSessions.LiveSessionLog') }
    static async getLookupAsync() { return getLookupAsync<LiveSessionLogRow>('LiveSessions.LiveSessionLog') }

    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<LiveSessionLogRow>();
}

File: GXpert/GXpert.Web/Modules/LiveSessions/LiveSessionLog/LiveSessionLogRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using Serenity.Extensions.Entities;
using System;
using System.ComponentModel;

namespace GXpert.LiveSessions;

[ConnectionKey("Default"), Module("LiveSessions"), TableName("LiveSessionLog")]
[DisplayName("Live Session Log"), InstanceName("Live Session Log")]
[ReadPermission(PermissionKeys.LiveSessionsManagement.View)]
[ModifyPermission(PermissionKeys.LiveSessionsManagement.Modify)]
[LookupScript("LiveSessions.LiveSessionLog")]
public sealed class LiveSessionLogRow : LoggingRow<LiveSessionLogRow.RowFields>, IIdRow
{
    const string jTeacher = nameof(jTeacher);
    const string jClass = nameof(jClass);
    const string jSubject = nameof(jSubject);

    [DisplayName("Id"), Identity, IdProperty]
    public int? Id { get => fields.Id[this]; set => fields.Id[this] = value; }

    [DisplayName("Date"), NotNull]
    public DateTime? Date { get => fields.Date[this]; set => fields.Date[this] = value; }

    [DisplayName("Start Time"), NotNull]
    public DateTime? StartTime { get => fields.StartTime[this]; set => fields.StartTime[this] = value; }

    [DisplayName("End Time"), NotNull]
    public DateTime? EndTime { get => fields.EndTime[this]; set => fields.EndTime[this] = value; }

    [DisplayName("Teacher"), NotNull, ForeignKey("Teachers", "Id"), LeftJoin(jTeacher), TextualField(nameof(TeacherPrn))]
    [LookupEditor("Users.Teacher")]
    public int? TeacherId { get => fields.TeacherId[this]; set => fields.TeacherId[this] = value; }

    [DisplayName("Class"), NotNull, ForeignKey("Classes", "Id"), LeftJoin(jClass), TextualField(nameof(ClassTitle))]
    [LookupEditor("Syllabus.Class")]
    public int? ClassId { get => fields.ClassId[this]; set => fields.ClassId[this] = value; }

    [DisplayName("Subject"), NotNull, ForeignKey("Subjects", "Id"), LeftJoin(jSubject), TextualField(nameof(SubjectTitle))]
    [LookupEditor("Syllabus.Subject")]
    public int? SubjectId { get => fields.SubjectId[this]; set => fields.SubjectId[this] = value; }

    [DisplayName("Is Active"), DefaultValue(1)]
    public bool? IsActive { get => fields.IsActive[this]; set => fields.IsActive[this] = value; }

    [DisplayName("Teacher Prn"), Expression($"{jTeacher}.[PRN]")]
    public string TeacherPrn { get => fields.TeacherPrn[this]; set => fields.TeacherPrn[this] = value; }

    [DisplayName("Class Title"), Expression($"{jClass}.[Title]")]
    public string ClassTitle { get => fields.ClassTitle[this]; set => fields.ClassTitle[this] = value; }

    [DisplayName("Subject Title"), Expression($"{jSubject}.[Title]")]
    public string SubjectTitle { get => fields.SubjectTitle[this]; set => fields.SubjectTitle[this] = value; }

    public class RowFields : LoggingRowFields
    {
        public Int32Field Id;
        public DateTimeField Date;
        public DateTimeField StartTime;
        public DateTimeField EndTime;
        public Int32Field TeacherId;
        public Int32Field ClassId;
        public Int32Field SubjectId;
        public BooleanField IsActive;

        public StringField TeacherPrn;
        public StringField ClassTitle;
        public StringField SubjectTitle;
    }
}